### PR TITLE
[New Rule] Attempt to Disable Gatekeeper

### LINKED ISSUE #420: [New Rule] Attempt to Disable Gatekeeper
## Description

Detects attempts to disable Gatekeeper on macOS. Gatekeeper is a security feature that's designed to ensure that only trusted software is run. Adversaries may attempt to disable Gatekeeper before executing malicious code.

```
$ sudo spctl --master-disable
$ sudo spctl --status
assessments disabled
$ sudo spctl --master-enable
$ sudo spctl --status
assessments enabled
```

`Security & Privacy` window with Gatekeeper disabled:

image

`Security & Privacy` window with Gatekeeper enabled:

image

## Required Info

- **Eventing Sources:**
`logs-endpoint.events.*`

- **Platforms**
`macOS`

## Optional Info
- **References:**
https://support.apple.com/en-us/HT202491

## Syntax
```
event.type:("start" or "process_started") and process.args:("spctl" and "--master-disable")
```

## Example Data
image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues
Resolves #420 

## Summary
Detects attempts to disable Gatekeeper on macOS. Gatekeeper is a security feature that's designed to ensure that only trusted software is run. Adversaries may attempt to disable Gatekeeper before executing malicious code.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+maturity = "production"
+updated_date = "2021/01/11"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects attempts to disable Gatekeeper on macOS. Gatekeeper is a security feature that's designed to ensure that only
+trusted software is run. Adversaries may attempt to disable Gatekeeper before executing malicious code.
+"""
+index = ["logs-endpoint.events.*"]
```
Comment (Samirbous) [inline rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml:12]: auditbeat ?

**Review by Samirbous — APPROVED** [review]

File: rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml (hunk 1 above)
Comment (bm11100) [inline rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml:12]:
```suggestion
index = ["auditbeat-*", "logs-endpoint.events.*"]
```

**Review by bm11100 — APPROVED** [review]
🐱‍🐉

File: rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "https://www.carbonblack.com/blog/tau-threat-intelligence-notification-new-macos-malware-variant-of-shlayer-osx-discovered/",
+]
+risk_score = 47
+rule_id = "4da13d6e-904f-4636-81d8-6ab14b4e6ae9"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Defense Evasion"]
+type = "query"
+
+query = '''
+event.type:("start" or "process_started") and process.args:("spctl" and "--master-disable")
```
Comment (brokensound77) [inline rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml:27]:
```suggestion
event.category:process and event.type:(start or process_started) and 
  process.args:(spctl and "--master-disable")
```

File: rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml (hunk 1 above)
Comment (threat-punter) [reply rules/macos/defense_evasion_attempt_to_disable_gatekeeper.toml:12]: 👌

**Pull request merged**